spec.bs

```diff
@@ -1364,6 +1414,25 @@ The <dfn attribute for="AppHistoryDestination">sameDocument</dfn> getter steps a
   1. For each |traversal| of |traversals|: [=finalize with an aborted navigation error=] given |appHistory| and |traversal|.
 </div>
 
+<div algorithm>
+  To <dfn>potentially reset the focus</dfn> given an {{AppHistory}} object |appHistory| and an {{AppHistoryNavigateEvent}} |event|:
+
+  1. Let |focusChanged| be |appHistory|'s [=AppHistory/focus changed during ongoing navigation=].
+  1. Set |appHistory|'s [=AppHistory/focus changed during ongoing navigation=] to false.
+  1. If |event|'s [=AppHistoryNavigateEvent/focus reset behavior=] is "{{AppHistoryFocusReset/manual}}", then return.
+     <p class="note">If it was left as null, then we treat that was "{{AppHistoryFocusReset/after-transition}}", and continue onward.
```

Comment:
This is not quite correct, because we need to only reset the focus if transitionWhile was actually called.